qemu/tests/qemu_disk_img.py

```diff
@@ -179,25 +177,37 @@ def verify_info(self, params=None):
                 raise error.TestFail(msg)
 
     @error_context.context_aware
-    def check_backingfile(self):
-        error_context.context("check image('%s') backing file" %
+    def get_backingfile(self):
+        """
+        get image backingfile
+
+        :return: Value of backing file, string or None;
+        """
+        error_context.context("get image('%s') backing file" %
                               self.image_filename, logging.info)
         out = self.get_info()
         try:
             backingfile = re.search(r'backing file: +(.*)', out, re.M).group(1)
-            if not self.base_tag or self.base_tag == "null":
-                msg = ("Expected backing file is null")
-                msg += " Actual backing file: %s" % backingfile
-                raise exceptions.TestFail(msg)
-            elif backingfile != self.base_image_filename:
-                msg = ("Expected backing file: %s" % self.base_image_filename)
-                msg += " Actual backing file: %s" % backingfile
-                raise exceptions.TestFail(msg)
+            return backingfile
         except AttributeError:
-            if self.base_tag and self.base_tag != "null":
-                msg = ("Could not find backing file for image '%s'" %
-                       self.image_filename)
-                raise exceptions.TestFail(msg)
+            return None
+
+    @error_context.context_aware
+    def check_backingfile(self):
+        """
+        check image backingfile
+        """
+        error_context.context("check image('%s') backing file" %
+                              self.image_filename, logging.info)
+        if not hasattr(self, "base_image_filename"):
+            expect = None
+        else:
+            expect = self.base_image_filename.strip("'")
```

Comment:
so that, you check backing file should have another arg  "expect_backingfile", you expect backingfile should include what you need.